django_pgschemas/middleware.py

```diff
@@ -89,6 +94,16 @@ def middleware(request):
         set_urlconf(urlconf)
 
         activate(tenant)
-        return get_response(request)
+
+    if iscoroutinefunction(get_response):
+        async_logic = sync_to_async(logic)
+
+        async def middleware(request):
+            await async_logic(request)
+            return await get_response(request)
+    else:
+        def middleware(request):  # type: ignore
+            logic(request)
```

Comment:
Please double check the logic here. `logic` can also return a redirection (in the middle of the function body) and therefore the function is not extractable as you have it.

Reply:
The failing tests are for this reason.

Reply:
@lorinkoz thanks for reviewing, I'm unable to run all the tests unfortunately, I get an `AppRegistryNotReady`. I've just pushed a change that should handle the redirect, however.